pkg/relayer/config/relayminer_configs_reader.go

```diff
@@ -37,37 +38,66 @@ func ParseRelayMinerConfigs(configContent []byte) (*RelayMinerConfig, error) {
 		return nil, ErrRelayMinerConfigUnmarshalYAML.Wrapf("%s", err)
 	}
 
+	// TODO_IN_THIS_COMMIT: fix comment...
+	// Check if the --query-node flag was set and use it if so.
 	// Check that the query node URL is provided
-	if yamlRelayMinerConfig.QueryNodeUrl == "" {
+	queryNodeUrlString := viper.GetString(KeyQueryNodeUrl)
+	switch {
+	case queryNodeUrlString != "":
```

Comment:
NIT: I think we shold leave the yaml reading function untouched and make it more functional.

```go
yamlRelayMinerConfigs = reader.ParseRelayMinerYamlConfigs(configContents)
relayMinerConfigs = reader.ViperConfigOverride(yamlRelayMinerConfigs)
```

That way things we explicitly mention via cli always take precedence.

Reply:
💯